keys-controller.js

```diff
@@ -60,9 +60,16 @@ function factory($scope, $routeParams, brAlertService, brKeyService,
   };
 
   self.init = function(identity) {
+    self.operations = {
+      add: false,
+      remove: false
+    };
     var sessionPromise = brSessionService.get().then(function(session) {
       self.isOwner = (session.identity &&
         (identity.id == session.identity.id));
+      if(self.isOwner || hasPermission(session.identity, 'PUBLIC_KEY_CREATE')) {
+        self.operations.add = true;
+      }
```

Comment:
I don't think this is the right place to do this. There should be an option to show the owner field -- and it should be controlled by an attribute on the component. Any session, identity, or permission checking code should be done outside of this module entirely -- but it will be used to provide the value for that attribute. We're adding too many dependencies to this module -- it shouldn't have all of these smarts baked in. We want to keep components simple -- with clear one-way binding inputs that control which features they expose.